### ISSUE TITLE
Adopting newer tf.random API?

### ISSUE THREAD

**Issue by sebastian-sz** [keras-team/keras-cv#155]
Most of [current layers](https://github.com/keras-team/keras-cv/search?q=tf.random) are using `tf.random` ops. From Tensorlfow's [RNG tutorial](https://www.tensorflow.org/guide/random_numbers):

> The old RNGs from TF 1.x such as [tf.random.uniform](https://www.tensorflow.org/api_docs/python/tf/random/uniform) and [tf.random.normal](https://www.tensorflow.org/api_docs/python/tf/random/normal) are not yet deprecated but strongly discouraged.

Are there plans to migrate `tf.random` ops to newer API of `tf.random.Generator` or `tf.random.stateless` ops?

### Context
As far as I understand: the newer API focuses more on reproducibility. For example:

```python
# OLD rng
for _ in range(2):
    for _ in range(3):
        print(tf.random.uniform((3, ), seed=1234))
```
will yield 6 different values, while
```python
# NEW rng
for _ in range(2):
    rng = tf.random.Generator.from_seed(1234)
    for _ in range(3):
        print(rng.uniform((1,)))
```
will output, the same values twice.

### How would this change the API?
Tensorflow mentions two ways of producing random numbers: either by using `tf.random.Generator` or by `tf.random.stateless_` ops. 
The former seems easier, as one could perhaps replace `tf.random` with `rng` calls. The latter would require to explicitly create and pass `seed` argument, which could be a bit more complicated but is still doable.

This could also probably make all random layers accept `seed` attribute, that would affect random behavior.

### Conclusions:
What are your thoughts on adopting this style of random numbers generation?

**Comment by LukeWood** [comment]
In our BaseAugmentation layer @qlzh727 included a RNG.  We should make use of the RNG he created in the base layer.

https://github.com/keras-team/keras/blob/d4480e8525440de6211c57c8df6c85d1aa168085/keras/engine/base_layer.py#L3424

`self._random_generator`

**Comment by LukeWood** [comment]
This falls under the same effort as migrating to the new `BaseImageAugmentationLayer` API

**Comment by sebastian-sz** [comment]
@LukeWood got it, thanks!
